app/controllers/api/v0/timelines/popular_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class Api::V0::Timelines::PopularController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :read, :'read:statuses' }, only: [:show]
+  before_action :require_user!, only: [:show]
+  after_action :insert_pagination_headers, unless: -> { @statuses.empty? }
+
+  def show
+    @statuses = sorted_statuses
+
+    render json: @statuses,
+           each_serializer: REST::StatusSerializer,
+           relationships: StatusRelationshipsPresenter.new(@statuses, current_user&.account_id),
+           status: account_home_feed.regenerating? ? 206 : 200
+  end
+
+  private
+
+  def sorted_statuses
+    load_statuses.sort_by(&:popularity)
+  end
+
+  def load_statuses
+    cached_home_statuses
+  end
+
+  def cached_home_statuses
+    cache_collection home_statuses, Status
+  end
+
+  def home_statuses
+    account_home_feed.get(
```

Comment:
Are we sure this will work how we want it to? If I'm understanding correctly, we're grabbing a page of Home and then sorting it by its popularity metric. If we allow the user to paginate, then they may see some popularity 'ramping' behaviour, where each new page/block of posts loaded is sorted individually amongst themselves, i.e., each batch of posts is sorted from most-to-least popular, rather than the whole feed itself.

Unless this was decided to be the case, should we instead be taking the last day or so of posts and sorting them in one go? (ala reddit top 24hr feed) We'd then paginate on that query instead.